Makefile

```diff
@@ -195,7 +195,7 @@ distclean: clean
 	$(MAKE) -C tools distclean
 
 setup:
-	$(MAKE) -C tools -j
+	$(MAKE) -C tools -j 4
```

Comment:
Two reasons:
1.  Should help with #597 and 
2. Its all I trust my pi with before it crashes.
But mostly the first one.

Reply:
btw, doing this produces the following warning:
`make[2]: warning: -j4 forced in submake: resetting jobserver mode.`

What if we remove every hardcoded `-j`? Running `make setup -j` doesn't do the same?

Reply:
I don't think we should use a specific number here. It'd be good to know what's crashing the pi to just use -j

Reply:
Passing -j without a specific number means that there wouldn't be any limit on the number of jobs, maybe it's too much for a Raspberry Pi.

Instead of doing that, may I suggest you to add `MAKEFLAGS += -j $(shell nproc)` within the platform detection code? 

e.g.:
```
    UNAME_S := $(shell uname -s)
    ifeq ($(UNAME_S),Linux)
        DETECTED_OS=linux
        MAKEFLAGS += -j $(shell nproc)
    endif
```

`nproc` is an useful utility belonging to GNU coreutils, thus it is commonly found on Linux systems. As for its function, [according to its manual](https://man7.org/linux/man-pages/man1/nproc.1.html), it automatically prints the number of cores available to the current process, this can be used to set the appropriate number of jobs for every system. For example, it would set 4 jobs on a Raspberry Pi that has four cores, but on a desktop with i7 or Ryzen it would set 8 (or more) jobs instead.

Reply:
I'm gonna resolve this cause we no longer specify -j at all here